### ISSUE TITLE
UMD - include params like window & document for better compression

### ISSUE THREAD

**Issue by alvarotrigo** [rollup/rollup#4395]
I'm building a front-end library.
I'm using the UDM module format and right now it looks like this:

```js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
    typeof define === 'function' && define.amd ? define(factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, global.demo = factory());
})(this, (function () { 'use strict';
```

However, I would like to include on the function params the `window` and `document` global variables.

Something like this:

```js
(function (global, factory, document, window) {
    typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory() :
    typeof define === 'function' && define.amd ? define(factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, global.demo = factory());
})(this, factory, document, window(function (document, window) { 'use strict';
    'use strict';
```

The main reason for that is to get a smaller output size when compressing it through Tester or any other plugin.
Adding these variables to the closure function allows them to get minified.
So, if I have a bundle file with 500 repetitions of the word `document`, that's 4Kb of data that can be reduced to 500bytes using a single word.

The same applies to `window` or any other global variable such as `Math`.

Any way to add these to the UMD module declaration?

This is my current config:

```js
 output: [
            {
                file: "build/demo.js",
                name: "demo",
                format: "umd",
                globals: {
                    'document': 'document',
                    'window': 'window'
                },
            },
```

**Comment by levrik** [comment]
I'm hitting this problem as well while trying to build a bundle for IE 11 which doesn't support `global`.

**Comment by lukastaegert** [comment]
If you also import window from a fake "window" module and declare that as external, that would solve your issue, wouldn't it?

https://rollupjs.org/repl/?version=3.5.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMHdpbmRvdyUyMGZyb20lMjAnd2luZG93JyUzQiU1Q24lNUNud2luZG93LmNvbnNvbGUubG9nKCdhd2Vzb21lJyklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJ1bWQlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlMjJ3aW5kb3clMjIlM0ElMjJ3aW5kb3clMjIlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

so basically

```
{
  // ...
  external: ['window'],
  output: {
    globals: {
      window': 'window'
    },
  }
}
```

and in your code

```
import window from 'window';
// ... whatever
```

You could even have a plugin using a `transform` function to inject that line into every file automatically.
